src/compiler/commandLineParser.ts

```diff
@@ -884,13 +888,306 @@ namespace ts {
+        /**
+         * Notify when unknown root key value option is being set
+         * @param key option key
+         * @param keyNode node corresponding to node in the source file
+         * @param value computed value of the key
+         * @param ValueNode node corresponding to value in the source file
+         */
+        onSetUnknownOptionKeyValueInRoot(key: string, keyNode: PropertyName, value: CompilerOptionsValue, valueNode: Expression): void;
+    }
+
+    /**
+     * Convert the json syntax tree into the json value
+     */
+    export function convertToObject(sourceFile: JsonSourceFile, errors: Diagnostic[]): any {
+        return convertToObjectWorker(sourceFile, errors, /*knownRootOptions*/ undefined, /*jsonConversionNotifier*/ undefined);
+    }
+
+    /**
+     * Convert the json syntax tree into the json value
+     */
+    function convertToObjectWorker(
+        sourceFile: JsonSourceFile,
+        errors: Diagnostic[],
+        knownRootOptions: Map<CommandLineOption> | undefined,
+        jsonConversionNotifier: JsonConversionNotifier | undefined): any {
+        if (!sourceFile.jsonObject) {
+            return {};
+        }
+
+        return convertObjectLiteralExpressionToJson(sourceFile.jsonObject, knownRootOptions,
+            /*extraKeyDiagnosticMessage*/ undefined, /*parentOption*/ undefined);
+
+        function convertObjectLiteralExpressionToJson(
+            node: ObjectLiteralExpression,
+            knownOptions: Map<CommandLineOption> | undefined,
+            extraKeyDiagnosticMessage: DiagnosticMessage | undefined,
+            parentOption: string | undefined
+        ): any {
+            const result: any = {};
+            for (const element of node.properties) {
+                if (element.kind !== SyntaxKind.PropertyAssignment) {
+                    errors.push(createDiagnosticForNodeInSourceFile(sourceFile, element, Diagnostics.Property_assignment_expected));
+                    continue;
+                }
+
+                if (element.questionToken) {
+                    errors.push(createDiagnosticForNodeInSourceFile(sourceFile, element.questionToken, Diagnostics._0_can_only_be_used_in_a_ts_file, "?"));
+                }
+                if (!isDoubleQuotedString(element.name)) {
+                    errors.push(createDiagnosticForNodeInSourceFile(sourceFile, element.name, Diagnostics.String_literal_with_double_quotes_expected));
+                }
+
+                const keyText = getTextOfPropertyName(element.name);
+                const option = knownOptions ? knownOptions.get(keyText) : undefined;
+                if (extraKeyDiagnosticMessage && !option) {
+                    errors.push(createDiagnosticForNodeInSourceFile(sourceFile, element.name, extraKeyDiagnosticMessage, keyText));
+                }
+                const value = convertPropertyValueToJson(element.initializer, option);
+                if (typeof keyText !== undefined && typeof value !== undefined) {

```

Comment:
This was merged like that??

Looks like an overlooked bug.

https://github.com/Microsoft/TypeScript/blob/master/src/compiler/commandLineParser.ts#L1056

Surely it was meant to be `typeof keyText !== 'undefined'` and same for value.